AISKU/src/Initialization.ts

```diff
@@ -241,23 +241,24 @@ export class Initialization implements IApplicationInsights {
      * @returns {IApplicationInsights}
      * @memberof Initialization
      */
-    public loadAppInsights(legacyMode: boolean = false): IApplicationInsights {
+    public loadAppInsights(legacyMode: boolean = false, core?: AppInsightsCore, sender?:Sender): IApplicationInsights {
 
         // dont allow additional channels/other extensions for legacy mode; legacy mode is only to allow users to switch with no code changes!
         if (legacyMode && this.config.extensions && this.config.extensions.length > 0) {
             throw new Error("Extensions not allowed in legacy mode");
         }
 
+        this.core = core || new AppInsightsCore();
         let extensions = [];
-        let appInsightsChannel: Sender = new Sender();
+        let appInsightsChannel: Sender = sender || new Sender();
 
         extensions.push(appInsightsChannel);
         extensions.push(this.properties);
         extensions.push(this.dependencies);
         extensions.push(this.appInsights);
 
         // initialize core
-        this.core.initialize(this.config, extensions);
+        core || this.core.initialize(this.config, extensions);
```

Comment:
`core.initialize` should be idempotent, so I don't think this is needed. Also, is the core you are passing in required to be already initialized?

Reply:
Hmm, not sure if it's idempotent or not. Since I was initializing the core already, I skipped it the second time.
But yes, core has to be initialized with the earlier channel sender

Reply:
Apparently you were right and calling it again didn't matter - updated